### ISSUE TITLE
Allowing fetching MAUDE-only MSIDs

### ISSUE THREAD

**Issue by taldcroft** [sot/eng_archive#137]
MAUDE has a lot of custom MSIDs that are not in the TDB.  One should be able to fetch those if `maude` is in the specified `data_source` list.

**Comment by jeanconn** [comment]
Now I'm just confused though, and wonder if this is a documentation problem.  It looks like the fetch works fine for me if I use my local maude with the new MSIDIDX file

```
ska-fido: unsetenv PYTHONPATH
ska-fido: ipython
Python 2.7.9 |Continuum Analytics, Inc.| (default, Apr 14 2015, 12:54:25) 
Type "copyright", "credits" or "license" for more information.

IPython 3.1.0 -- An enhanced Interactive Python.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://binstar.org
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from Ska.engarchive import fetch

In [2]: fetch.data_source.set('maude')

In [3]: fetch.Msid("ACIMG0K1", '2017:001:00:00:00.000', '2017:001:00:01:00.000').vals
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-055ab2a565c5> in <module>()
----> 1 fetch.Msid("ACIMG0K1", '2017:001:00:00:00.000', '2017:001:00:01:00.000').vals

/proj/sot/ska/arch/x86_64-linux_CentOS-5/lib/python2.7/site-packages/Ska.engarchive-3.41.3-py2.7.egg/Ska/engarchive/fetch.pyc in __init__(self, msid, start, stop, filter_bad, stat)
   1650     def __init__(self, msid, start=LAUNCH_DATE, stop=None, filter_bad=True, stat=None):
   1651         super(Msid, self).__init__(msid, start=start, stop=stop,
-> 1652                                    filter_bad=filter_bad, stat=stat)
   1653 
   1654 

/proj/sot/ska/arch/x86_64-linux_CentOS-5/lib/python2.7/site-packages/Ska.engarchive-3.41.3-py2.7.egg/Ska/engarchive/fetch.pyc in __init__(self, msid, start, stop, filter_bad, stat)
    502 
    503     def __init__(self, msid, start=LAUNCH_DATE, stop=None, filter_bad=False, stat=None):
--> 504         msids, MSIDs = msid_glob(msid)
    505         if len(MSIDs) > 1:
    506             raise ValueError('Multiple matches for {} in Eng Archive'

/proj/sot/ska/arch/x86_64-linux_CentOS-5/lib/python2.7/site-packages/Ska.engarchive-3.41.3-py2.7.egg/Ska/engarchive/fetch.pyc in msid_glob(msid)
    394     if not msids:
    395         raise ValueError('MSID {!r} is not in {} data source(s)'
--> 396                          .format(msid, ' or '.join(x.upper() for x in sources)))
    397 
    398     return list(msids), list(MSIDS)

ValueError: MSID 'ACIMG0K1' is not in MAUDE data source(s)

In [4]: 
Do you really want to exit ([y]/n)? y
ska-fido: setenv PYTHONPATH ~/git/maude
ska-fido: ipython
Python 2.7.9 |Continuum Analytics, Inc.| (default, Apr 14 2015, 12:54:25) 
Type "copyright", "credits" or "license" for more information.

IPython 3.1.0 -- An enhanced Interactive Python.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://binstar.org
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from Ska.engarchive import fetch

In [2]: fetch.data_source.set('maude')

In [3]: fetch.Msid("ACIMG0K1", '2017:001:00:00:00.000', '2017:001:00:01:00.000').vals
Out[3]: array([10, 10, 10, 10, 10, 10,  9, 10, 10, 10,  9,  9, 10, 10, 10], dtype=uint16)

In [4]: 
```

**Comment by taldcroft** [comment]
Great!  I assumed you had originally done this test and I didn't actually check in the code to see if this would work.

**Comment by jeanconn** [comment]
Right, I think you got my bug reports out of order.  All good.  First I tried fetch, did not work and I basically reported that inferred that might be a Ska.tdb thing (because I also saw these msids weren't in there and I thought maybe there was a dependence), but then saw that these msids didn't fetch in `maude` directly, updated that, and then got back to checking fetch with it by the time you were opening this issue.
